qiskit/circuit/library/arithmetic/linear_amplitude_function.py

```diff
@@ -102,8 +103,10 @@ def __init__(
         """
         if not hasattr(slope, "__len__"):
             slope = [slope]
+        slope = cast(List[float], slope)
```

Comment:
Similar comment as https://github.com/Qiskit/qiskit-terra/pull/8267#discussion_r1101205233: since we don't run mypy at the moment, I think it would be better to not add `cast`s as new code won't have it and we don't check it right now 🙂